src/index.js

```diff
@@ -1,11 +1,11 @@
 const slugify = (str) => {
   return str
     .toLowerCase()
-    .replace(/\s/g, "-") // Replace white space
-    .replace(/[^a-zA-Z0-9-_]/g, "-") // Replace non alphanum-_ chars
-    .replace(/-{2,}/g, "-") // Remove extra - occurences
-    .replace(/-$/g, "") // Remove - at the begening of slug
-    .replace(/^-/g, ""); // Remove - at the end of slug
+    .replace(new RegExp("\\s", "g"), "-") // Replace white space
+    .replace(new RegExp("[^a-zA-Z0-9-_]", "g"), "-") // Replace non alphanum-_ chars
+    .replace(new RegExp("-{2,}", "g"), "-") // Remove extra - occurences
+    .replace(new RegExp("-$", "g"), "") // Remove - at the beginning of slug
+    .replace(new RegExp("^-", "g"), ""); // Remove - at the end of slug
```

Comment:
Using the Regex object here don't bring any benefit, and even comes with a downside since the regex evaluation will be done at runtime instead of build time.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp

```
The literal notation results in compilation of the regular expression when the expression is evaluated. Use literal notation when the regular expression will remain constant.
...
The constructor of the regular expression object—for example, new RegExp('ab+c')—results in runtime compilation of the regular expression. Use the constructor function when you know the regular expression pattern will be changing
```

Please rollback this part

Reply:
Done